File: test-workspaces/modules/shared_common/taxonomy_dup_processor_b.py
# Duplicated processing logic for test workspace
# This file contains duplicate code to trigger AES305 detection


def process_data(input_text):
    trimmed = input_text.strip()
    normalized = trimmed.lower()
    result = ''.join(c for c in normalized if c.isalnum())
    processed = result
    final_result = processed.replace("  ", " ")
    return final_result


def format_output(data):
    return f"[PROCESSED] {data}"


def validate_input(input_text):
    return not input_text.strip() == ""


def transform_data(data):
    return [line.strip() for line in data.splitlines() if line.strip()]

File: test-workspaces/crates/shared_common/src/taxonomy_dup_processor_b.rs
// Duplicated processing logic for test workspace
// This file contains duplicate code to trigger AES305 detection

pub fn process_data(input: &str) -> String {
    let trimmed = input.trim();
    let normalized = trimmed.to_lowercase();
    let result: Vec<char> = normalized.chars().filter(|c| c.is_alphanumeric()).collect();
    let processed = result.into_iter().collect::<String>();
    let final_result = processed.replace("  ", " ");
    final_result
}

pub fn format_output(data: &str) -> String {
    format!("[PROCESSED] {}", data)
}

pub fn validate_input(input: &str) -> bool {
    !input.trim().is_empty()
}

pub fn transform_data(data: &str) -> Vec<String> {
    data.lines().map(|l| l.trim().to_string()).filter(|l| !l.is_empty()).collect()
}
